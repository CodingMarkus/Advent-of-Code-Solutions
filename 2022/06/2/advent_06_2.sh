#!/bin/sh

# Run as:
# cat advent_06_2_sample.txt | sh advent_06_2.sh
# cat advent_06_2_input.txt  | sh advent_06_2.sh

set -e

seq=$( dd bs=1 count=14 2>/dev/null )
[ -n "$seq" ] || { echo "Error 1"; exit 1; }
[ ${#seq} -eq 14 ] || { echo "Error 2"; exit 1; }

readonly true=0
readonly false=1

hasDupes()
{
	cnt=$( printf '%s' "$1" | sed 's/\(.\)/\1\n/g' | sort | uniq | wc -l )
	[ "$cnt" -eq 14 ] && return $false
	return $true
}


count=14
read -r line || true

for char in $( printf '%s' "$line" | sed 's/\(.\)/\1 /g' )
do
	if ! hasDupes "$seq"
	then
		echo "Count: $count"
		exit 0
	fi

	count=$(( count + 1 ))
	seq=$( printf '%s' "$seq" | cut -c 2- )
	seq="$seq$char"
done

echo "Not found"
exit 1