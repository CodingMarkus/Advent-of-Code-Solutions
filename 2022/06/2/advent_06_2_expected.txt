Count: 19
Count: 2122
